samza-core/src/main/java/org/apache/samza/config/BlobStoreConfig.java

```diff
@@ -24,18 +24,59 @@
  */
 public class BlobStoreConfig extends MapConfig {
 
-  public static final String BLOB_STORE_MANAGER_FACTORY = "blob.store.manager.factory";
-  public static final String BLOB_STORE_ADMIN_FACTORY = "blob.store.admin.factory";
+  private static final String PREFIX = "blob.store.";
+  public static final String BLOB_STORE_MANAGER_FACTORY = PREFIX + "manager.factory";
+  public static final String BLOB_STORE_ADMIN_FACTORY = PREFIX + "admin.factory";
+  // Configs related to retry policy of blob stores
+  private static final String RETRY_POLICY_PREFIX = PREFIX + "retry.policy.";
+  public static final String RETRY_POLICY_MAX_RETRIES = RETRY_POLICY_PREFIX + "max.retries";
+  // -1 for RetryPolicy means unlimited retries. Retry is limited by max retry duration, rather than count of retries.
+  public static final int DEFAULT_RETRY_POLICY_MAX_RETRIES = -1;
+  public static final String RETRY_POLICY_MAX_RETRY_DURATION_MILLIS = RETRY_POLICY_PREFIX + "max.retry.duration.millis";
+  public static final long DEFAULT_RETRY_POLICY_MAX_RETRY_DURATION_MILLIS = 10 * 60 * 1000; // 10 mins
```

Comment:
Let's change this to > 10 min. IIRC the current policy defaults only try up to 6-ish minutes and the last attempt exceeds this 10 min max.

Reply:
Thanks, updating it to 15 mins. Given max delay of 5 mins, that should take > 10 mins to exhaust all retries.